packages/ui-top-nav-bar/src/TopNavBar/TopNavBarLayout/props.ts

```diff
@@ -49,83 +47,163 @@ import type { BrandChild } from '../TopNavBarBrand/props'
 import type { MenuItemsChild } from '../TopNavBarMenuItems/props'
 import type { UserChild } from '../TopNavBarUser/props'
 
+import { topNavBarItemTooltipPropType } from '../TopNavBarItem/props'
+
 import { TopNavBarLayout } from './index'
+import type { DesktopLayoutOwnProps } from './DesktopLayout/props'
+import type { SmallViewportLayoutOwnProps } from './SmallViewportLayout/props'
 
 type LayoutChild = React.ComponentElement<TopNavBarLayoutProps, TopNavBarLayout>
 
-type TopNavBarLayoutOwnProps = {
+type CommonTopNavBarLayoutProps = {
   /**
-   * TODO: desc
+   * Displays the app/product/brand/company/etc. name and/or logo.
+   *
+   * Accepts a `<TopNavBar.Brand>` component.
    */
   renderBrand?: BrandChild
 
   /**
-   * TODO: desc
+   * Displays the main navbar items.
+   *
+   * In __desktop__ mode the items are listed on the navbar,
+   * in __smallViewport__ mode the items are accessible under the main "hamburger" menu.
+   *
+   * Accepts a `<TopNavBar.MenuItems>` component.
    */
   renderMenuItems?: MenuItemsChild
 
   /**
-   * TODO: desc
+   * Displays the action items, icons, buttons, etc.
+   *
+   * Renders in the top right corner of the navbar in both __desktop__ and __smallViewport__ mode.
+   *
+   * Accepts a `<TopNavBar.ActionItems>` component.
    */
   renderActionItems?: ActionItemsChild
 
   /**
-   * TODO: desc
+   * Displays the user menu.
+   *
+   * In __desktop__ mode it renders in the top right corner of the navbar,
+   * in __smallViewport__ mode it is rendered under the main "hamburger" menu, above the menu items.
+   *
+   * Accepts a `<TopNavBar.User>` component.
    */
   renderUser?: UserChild
 
   /**
-   * TODO: desc - it is used for the small viewport layout!
+   * The 'aria-label' for the underlying `<nav>` element
    */
-  trayMountNode?: TrayProps['mountNode']
+  navLabel?: string
 
   /**
    * A function that returns a reference to root HTML element
    */
-  elementRef?: (el: Element | null) => void
+  elementRef?: (el: HTMLElement | null) => void
+}
+
+type TopNavBarLayoutOwnProps = CommonTopNavBarLayoutProps & {
+  /**
+   * Config object for the "desktop" mode:
+   */
+  desktopConfig?: DesktopLayoutOwnProps
+
+  /**
+   * Config object for the "small viewport" mode. It is required, if there is no custom `renderSmallViewportLayout` is set on `<TopNavBar>`.
```

Comment:
There is no `renderSmallViewportLayout` prop on the `TopNavBar` component, and there seems to be no related code in the package, maybe this is an outdated comment or the functionality is missing?

Reply:
Thanks, it was outdated, fixed!